content/get-started/getting-started-with-git/configuring-git-to-handle-line-endings.md

```diff
@@ -107,29 +107,40 @@ You'll notice that files are matched—`*.c`, `*.sln`, `*.png`—, separated by
 
 When you set the `core.autocrlf` option or commit a *.gitattributes* file, you may find that Git reports changes to files that you have not modified. Git has changed line endings to match your new configuration.
 
-To ensure that all the line endings in your repository match your new configuration, backup your files with Git, delete all files in your repository (except the `.git` directory), then restore the files all at once.
-
-1. Save your current files in Git, so that none of your work is lost.
-  ```shell
-  $ git add . -u
-  $ git commit -m "Saving files before refreshing line endings"
-  ```
-2. Add all your changed files back and normalize the line endings.
-  ```shell
-  $ git add --renormalize .
-  ```
-3. Show the rewritten, normalized files.
-  ```shell
-  $ git status
-  ```
-4. Commit the changes to your repository.
-  ```shell
-  $ git commit -m "Normalize all the line endings"
-  ```
+To ensure that all the line endings in your repository match your new configuration, backup your files with Git, then remove/restore the files all at once with renormalized line endings.
+
+1. Save your current files in Git, so that none of your work is lost.  
+   ```shell
+   $ git add . -u
+   $ git commit -m "Saving files before refreshing line endings"
+   ```
+2. Add/re-add any new/changed files and normalize the line endings.  
+   ```shell
+   $ git add --renormalize .
+   ```  
+   Or, alternatively, if you just need to convert all files in the HEAD of an existing checked-out repository to the new git line-ending configuration:  
```

Comment:
When possible, we like to provide one procedure for accomplishing a goal. If we can't provide one set of steps that applies to every situation, I'd like to start this step by detailing the two options, and why you might choose one over the other.

@stephenskett, could you provide that context here? How would someone know which approach they'd want to take? Once I'm clearer on that, I'll restructure this a bit 👀

Reply:
Hi Matt, here's my best attempt to identify the respective contexts for the two approaches I mentioned in my adjusted version.

Firstly, I feel I should probably clarify up front that I am a relative newbie to both git & GitHub. So, whilst I started this PR because I couldn't get the previous instructions to work as expected, it's entirely possible I've misunderstood the intended scope of the article as currently constituted. Hence, my changes are absolutely a work in progress and so, of course, feel free to reword/restructure as you see fit!

I think the basic problem I was having is, there are two separate scenarios where you might want to use git to 'refresh' your repo's line endings after adjusting the config/.gitattributes:
1. Git detects changes in unedited repo files due to new line-ending config;
2. You just want to convert all files in the current working dir (HEAD) to have the line endings as per the repo config, irrespective of whether git status/diff detects any change to the files.

As I re-read the original version of the article, I sense maybe it is only really dealing with situation 1 - is that correct? Whereas, my use-case would fall under point 2. I naively assumed it would work for both, but it didn't: the line `git add --renormalize .` & subsequent `commit` just didn't do anything in my case. Presumably this is because, Git didn't actually detect any diff's and so as far as it could tell there was nothing that needed changing.

I appreciate there are other ways of dealing with my situation, rather than using Git to do the conversion (e.g. using an IDE to manage your project's line-endings; cloning & re-checking-out the repo to update the line endings in the HEAD), but I think those options aren't always possible, and/or desirable - sometimes you just want to update the HEAD of the local repo without necessarily needing to do a new commit, and I think my alternative method allows that.

I see now that my suggested change needs a bit more explanation!! So, perhaps it would make sense to delineate these two scenarios at the top of the section, to avoid others getting confused like I did? And/or, have separate sub-sections, with separately delineated instructions for each use-case?

Reply:
P.s. Happy to make further changes as required based on your follow-up comments. Or, if you'd rather make the changes yourself based on your wider scheme of restructuring the article, that'd be fine with me too. 😃

Reply:
Thank you 🙏🏻 Given that `git add --renormalize` and the subsequent `commit` did nothing for you, I'm wondering if we could actually avoid the "alternatively..." bit by combining the two procedures and having readers commit their changes, then use your commands to update all of the files on the current branch—which should include the changes they just committed, if there were any.

1. To avoid losing any local changes to files in the repository, add and commit any outstanding changes by running the following commands.

       git add . -u
       git commit -m "Saving files before refreshing line endings"
1. To ensure all files for the current branch are updated to reflect the repository's new settings for line endings, run the following commands.

       git rm -rf --cached .
       git reset --hard HEAD

I'm happy to do a little testing here, too. Let me know if that'd help 😄

Reply:
Heya, thanks for this suggestion and sorry for delayed response.
My only slight concern around combining the methods in this way with no further qualification is that, we probably need to make clear there is a slightly different set of risks associated with doing the step #2 in my method, cf. the original version - e.g. if the user has any binary files in their repo & uses a text-only repo-config/.gitattributes, the `git reset ...` line will most likely corrupt the binary files.

Sorry, I know it would probably make sense to have included that point in the original PR, but I couldn't think of where best to fit it in!

Hope that helps with your deliberations. 😉

Reply:
Just to try and keep the ball rolling on this: perhaps we could say something like this instead?

1. Verify that all configuration changes have been applied correctly, and, in particular, that any binary file-types in the repository are specified as such in *.gitattributes*.  
2. To ensure that none of your work is lost, add and commit any changes by running the following commands.  
   ```shell{:copy}
   git add . -u
   git commit -m "Saving files before refreshing line endings"
   ```
3. To ensure all files for the current branch are updated to reflect the repository's new settings for line endings, run the following commands.  
   ```shell{:copy}
   $ git rm -rf --cached .
   $ git reset --hard HEAD
   ```
4. Run `git status` to display any changed files after renormalization.  
   If no further changes to the repo are required, step 5 can be skipped.
5. To commit the changes to your repository (if required), run the following command.  
   ```shell{:copy}
   git commit -m "Normalize all the line endings"
   ```

Reply:
👋🏻 Thanks for your patience. This looks like a solid approach 🎉 Mind taking a look at my latest revision?

Reply:
This looks fantastic! Happy to resolve and move forward as suggested. Great work, and thanks so much for taking my suggestions into consideration. :satisfied::blush::smile: